### ISSUE TITLE
wrong length reject with rule attack (WPA hash)

### ISSUE THREAD

**Issue by Ali-00-Ridha** [hashcat/hashcat#2202]
the rule is :**T0 $2 $0 $0 $0**
the word is :**kadem**
the hccapx file: https://www.sendspace.com/file/diu7ll

the password of the hash in the hccapx file is (Kadem2000) so hashcat should be able to recover it with provided **rule + word** attack  but hashcat _reject_ the word _even so the password is longer than 8 chars_ (obviously)

**when I re-write the rule to be only (T0) and the word (kadem2000) then it works and it recovers the password**

so the problem obviously that hashcat checks the length of the word in the file and _**not**_ the result password _**after applying the rules**_ 

this behavour happens even with older versions of hashcat like _**5.0, 4.2.1, 3.6**_ where all have the same problem.

please fix it fast, Thank you :) for your hard work.

**Comment by Rdgeno** [comment]
What do you have -m set at that's probably your problem. Look at hashcat help to get the right one.

**Comment by Rdgeno** [comment]
-r best64.rule seems to use all lengths but refuses the short ones. I'm not 100% clear on what your talking about.

**Comment by Rdgeno** [comment]
Sorry I read your post wrong. Try the preset rules. If it doesnt work wlm a day without tape will append and prepend.

**Comment by Rdgeno** [comment]
Your word or the mentalist. Either will fix you until Hashcat gets to it. Dont hold your breath opencl is still an issue.

**Comment by Ali-00-Ridha** [comment]
I have posted this problem on hashcat forum and I got this answer:
rules are applied on GPU for performance reasons (after the fact, after the rejection routine) and we can't really filter each and every combination correctly on GPU (syncing this would be very complex and make everything very slow).

there are 2 known workarounds (might be much slower than without them):
1. use -S
2. use a pipe: hashcat --stdout -r a.rule dict.txt | hashcat -a 0 -w 3 -m 2500 my.hccapx

So I will close this issue.
